kotlin-asyncapi-context/src/main/kotlin/org/openfolder/kotlinasyncapi/context/annotation/AnnotationProvider.kt

```diff
@@ -70,16 +72,24 @@ class AnnotationProvider(
                 listOfNotNull(
                     clazz.findAnnotation<Message>()?.let { clazz to it },
                     clazz.findAnnotation<Schema>()?.let { clazz to it },
-                    clazz.findAnnotation<Channel>()?.let { clazz to it }
+                    clazz.findAnnotation<Channel>()?.let { clazz to it },
+                    clazz.findAnnotation<AsyncApiComponent>()?.let { clazz to it}
                 )
             }
             .mapNotNull { (clazz, annotation) ->
                 when (annotation) {
                     is Message -> messageProcessor.process(annotation, clazz)
                     is Schema -> schemaProcessor.process(annotation, clazz)
                     is Channel -> channelProcessor.process(annotation, clazz).also {
-                        componentToChannelMapping[clazz.java.simpleName] =
-                            annotation.value.takeIf { it.isNotEmpty() } ?: clazz.java.simpleName
+                        if (clazz.annotations.any { it is Channel }) {
+                            componentToChannelMapping[clazz.java.simpleName] =
+                                annotation.value.takeIf { it.isNotEmpty() } ?: clazz.java.simpleName
+                        }
+                    }
```

Comment:
Why is this if check needed? We already made sure that class is annotated with `@Channel` in the flatMap step before.